src/main/java/org/jenkinsci/test/acceptance/plugins/git/GitRepo.java

```diff
@@ -94,16 +94,22 @@ public void setIdentity(String userName, String userMail) {
     private File initDir() {
         try {
             // FIXME: perhaps this logic that makes it use a separate key should be moved elsewhere?
-            privateKey = File.createTempFile("ssh", "key");
-            FileUtils.copyURLToFile(GitContainer.class.getResource("GitContainer/unsafe"), privateKey);
-            Files.setPosixFilePermissions(privateKey.toPath(), singleton(OWNER_READ));
+            privateKey = Files.createTempFile("ssh", "key").toFile();
 
-            ssh = File.createTempFile("jenkins", "ssh");
-            FileUtils.writeStringToFile(ssh,
-                    "#!/bin/sh\n" +
-                            "exec ssh -o StrictHostKeyChecking=no -i " + privateKey.getAbsolutePath() + " \"$@\"" , StandardCharsets.UTF_8);
-            Files.setPosixFilePermissions(ssh.toPath(), new HashSet<>(Arrays.asList(OWNER_READ, OWNER_EXECUTE)));
+            FileUtils.copyURLToFile(GitContainer.class.getResource("GitContainer/unsafe"), privateKey);
 
+            if (SystemUtils.IS_OS_WINDOWS) {
+                ssh = Files.createTempFile("jenkins", "ssh.cmd").toFile();
+                // this works if ssh is on the path, for example
+                // https://learn.microsoft.com/en-us/windows-server/administration/openssh/openssh_install_firstuse
+                Files.writeString(ssh.toPath(), "ssh.exe -o StrictHostKeyChecking=no -i " + privateKey.getAbsolutePath() + " %*", Charset.defaultCharset());
```

Comment:
both the linux and windows implementations use the default charset as this is run by tooling that may be not using `UTF-8`